sentry/src/main/java/io/sentry/metrics/MetricsApi.java

```diff
@@ -0,0 +1,512 @@
+package io.sentry.metrics;
+
+import io.sentry.IMetricsAggregator;
+import io.sentry.MeasurementUnit;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+// TODO: add tons of method overloads to make it delightful to use
+public final class MetricsApi {
```

Comment:
As an alternative we could have a builder pattern instead, e.g.
```
Sentry.metrics().increment()
    .value(1.0)
    .tag("key", "value")
    .emit()
```

Reply:
as another alternative we could have 1 "simplified" method and 1 "full" method. e.g.

```
public void increment(final @NotNull String key) { ... }
public void increment(
    final @NotNull String key,
    final double value,
    final @Nullable MeasurementUnit unit,
    final @Nullable Map<String, String> tags,
    final @Nullable Long timestampMs,
    final int stackLevel) { ... }
```
wdyt?
Or just keep it as is, "consistently" with what we have in other places of the SDK

Reply:
let's keep it as-is for now and see if we get any feedback.